google-cloud-logging/lib/google/cloud/logging/service.rb

```diff
@@ -279,6 +285,24 @@ def default_options
           Google::Gax::CallOptions.new kwargs: default_headers
         end
 
+        ##
+        # @private Get a Google::Protobuf::Timestamp object from a Time object.
+        def time_to_timestamp time
+          return nil if time.nil?
+          # Make sure we have a Time object
+          return nil unless time.respond_to? :to_time
```

Comment:
If we were to raise something like ArgumentError the place to do it is in the public method that accepts the value, not this private conversion method.

Do you want to see an error raised if a non-Time object is provided?